src/sharedHooks/useAuthenticatedQuery.js

```diff
@@ -1,29 +1,60 @@
 import { useQuery } from "@tanstack/react-query";
-import { getJWT } from "components/LoginSignUp/AuthenticationService.ts";
+import { getJWT, isLoggedIn } from "components/LoginSignUp/AuthenticationService.ts";
+import { useEffect, useState } from "react";
+import { logWithoutRemote } from "sharedHelpers/logger";
 import { reactQueryRetry } from "sharedHelpers/logging";
 
+const localLogger = logWithoutRemote.extend( "useAuthenticatedQuery" );
+
+const LOGGED_IN_UNKNOWN = null;
+
 // Should work like React Query's useQuery except it calls the queryFunction
 // with an object that includes the JWT
 const useAuthenticatedQuery = (
   queryKey,
   queryFunction,
   queryOptions = {}
-) => useQuery( {
-  queryKey: [...queryKey, queryOptions.allowAnonymousJWT],
-  queryFn: async ( ) => {
-    // Note, getJWT() takes care of fetching a new token if the existing
-    // one is expired. We *could* store the token in state with useState if
-    // fetching from RNSInfo becomes a performance issue
-    const apiToken = await getJWT( queryOptions.allowAnonymousJWT );
-    const options = {
-      api_token: apiToken
-    };
-    return queryFunction( options );
-  },
-  retry: ( failureCount, error ) => reactQueryRetry( failureCount, error, {
-    queryKey
-  } ),
-  ...queryOptions
-} );
+) => {
+  const [userLoggedIn, setUserLoggedIn] = useState( LOGGED_IN_UNKNOWN );
+
+  // Whether we perform this query and whether we need to re-perform it
+  // depends on whether the user is signed in. The vulnerability here is that
+  // this effect might not run frequently enough to change when a user signs
+  // in or out
+  useEffect( ( ) => {
+    isLoggedIn()
+      .then( result => setUserLoggedIn( result ) )
+      .catch( ( ) => setUserLoggedIn( LOGGED_IN_UNKNOWN ) );
+  }, [queryKey, queryOptions] );
+
+  // The results will probably be different depending on whether the user is
+  // signed in or we wouldn't be using useAuthenticatedQuery in the first
+  // place, so we need to redo this request if the auth state changed
+  const authQueryKey = [...queryKey, queryOptions.allowAnonymousJWT, userLoggedIn];
+  if ( authQueryKey.includes( "useIconicTaxa" ) ) {
+    localLogger.debug( "authQueryKey: ", authQueryKey );
+  }
+
+  return useQuery( {
```

Comment:
Is this all necessary? I thought react-query was already handling this when you add more items into the query key, i.e. if the query key in `useIconicTaxa` had three+ items in the array: `const queryKey = ["useIconicTaxa", reload, queryOptions.allowAnonymousJWT, userLoggedIn]` then it would fetch, cache, and return different query results for a query key with `userLoggedIn = true` vs. `userLoggedIn = false` and with or without an anonymous jwt.

Reply:
It's not necessary to fix this bug, but I am anticipating other, similar bugs where we accidentally use results from an unauthenticated query when we thought we were using results from an authenticated one. So yes, I could have simply altered the key we pass from `useIconicTaxa` to `useAuthenticatedQuery` and that would have fixed MOB-383, but this solution guards against similar bugs in the future.